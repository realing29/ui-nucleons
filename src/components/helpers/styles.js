import isNumber from 'lodash/isNumber';

/**
 * Возвращает строку значения CSS-свойства "transition".
 * @param {number} [duration=0] Длительность перехода в миллисекундах.
 * @param {string} [property='all'] Имя свойства.
 * @param {string} [easing='ease-out'] Функция плавности.
 * @return {string} Значение CSS-свойства "transition".
 */
export const getTransitionStyle = (
  duration = 0,
  property = 'all',
  easing = 'ease-out'
) => `${property} ${isNumber(duration) ? duration : 0}ms ${easing}`;

/**
 * Возвращает строку значения CSS-свойства с трансформацией смещения.
 *
 * Использовать только "translate3d" для смещения.
 * @see https://stackoverflow.com/a/22312917
 *
 * @param {number} [x=0] Смещение по оси x в пикселях.
 * @param {number} [y=0] Смещение по оси y в пикселях.
 * @param {number} [z=0] Смещение по оси z в пикселях.
 * @return {string} Значение CSS-свойства с трансформацией смещения.
 */
export const getTranslateStyle = (x = 0, y = 0, z = 0) => `translate3d(${x}px, ${y}px, ${z}px)`;
