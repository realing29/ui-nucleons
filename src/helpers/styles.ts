/**
 * Возвращает строку значения CSS-свойства "transition".
 * @param duration Длительность перехода в миллисекундах.
 * @param property Имя свойства.
 * @param easing Функция плавности.
 * @return Значение CSS-свойства "transition".
 */
export function getTransitionStyle(duration = 0, property = 'all', easing = 'ease-out'): string {
  return `${property} ${Number.isFinite(duration) ? duration : 0}ms ${easing}`;
}

/**
 * Возвращает строку значения CSS-свойства с трансформацией смещения.
 *
 * Использовать только "translate3d" для смещения.
 * @see https://stackoverflow.com/a/22312917
 * @param x Смещение по оси x в пикселях.
 * @param y Смещение по оси y в пикселях.
 * @param z Смещение по оси z в пикселях.
 * @return Значение CSS-свойства с трансформацией смещения.
 */
export function getTranslateStyle(x = 0, y = 0, z = 0): string {
  return `translate3d(${x}px, ${y}px, ${z}px)`;
}

/**
 * Устанавливает на переданный элемент css-переменную "--vh".
 * Значение переменной является аналогом `1vh` с учетом особенностей viewport'а мобильный устройств.
 * @param element Элемент на котором будет задана css-переменная.
 */
export function setViewportHeightUnit(element: HTMLElement) {
  const height = visualViewport ? visualViewport.height * visualViewport.scale : window.innerHeight;

  element.style.setProperty('--vh', `${height / 100}px`);
}
