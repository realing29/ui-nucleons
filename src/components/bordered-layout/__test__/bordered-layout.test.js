import React from 'react';
import { shallow } from 'enzyme';
import BorderedLayout from '../';
import styles from '../bordered-layout.scss';

describe('<BoBorderedLayoutrder />', () => {
  it('without props', () => {
    const component = shallow(<BorderedLayout />);
    expect(component.find('div').at(0).hasClass(styles['disable-border-top'])).toBeTruthy();
    expect(component.find('div').at(0).hasClass(styles['disable-border-bottom'])).toBeTruthy();
    expect(component.find('div').at(1).hasClass(styles['disable-border-top'])).toBeTruthy();
    expect(component.find('div').at(1).hasClass(styles[['disable-border-bottom']])).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
  it('with props', () => {
    const component = shallow(
      <BorderedLayout
        top
        bottom
      />
    );
    expect(component.find('div')).toHaveLength(2);
    expect(component.find('div').at(0).hasClass(styles.base)).toBeTruthy();
    expect(component.find('div').at(0).hasClass(styles.outer)).toBeTruthy();
    expect(component.find('div').at(1).hasClass(styles.base)).toBeTruthy();
    expect(component.find('div').at(1).hasClass(styles.inner)).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
