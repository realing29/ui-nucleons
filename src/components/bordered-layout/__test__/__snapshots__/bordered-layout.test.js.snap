// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoBorderedLayoutrder /> with props 1`] = `
<div
  className="base outer "
>
  <MobileLayout>
    <div
      className="base inner "
    />
  </MobileLayout>
</div>
`;

exports[`<BoBorderedLayoutrder /> without props 1`] = `
<div
  className="base outer disable-border-top disable-border-bottom"
>
  <MobileLayout>
    <div
      className="base inner disable-border-top disable-border-bottom"
    />
  </MobileLayout>
</div>
`;
